File: app/src/main/java/com/lifedawn/bestweather/weathers/detailfragment/currentconditions/DetailCurrentConditionsFragment.java
package com.lifedawn.bestweather.weathers.detailfragment.currentconditions;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.lifedawn.bestweather.R;
import com.lifedawn.bestweather.weathers.detailfragment.base.BaseDetailCurrentConditionsFragment;
import com.lifedawn.bestweather.weathers.models.CurrentConditionsDto;

import org.jetbrains.annotations.NotNull;

public class DetailCurrentConditionsFragment extends BaseDetailCurrentConditionsFragment {
	private CurrentConditionsDto currentConditionsDto;


	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
	}

	@Override
	public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		setValuesToViews();
	}

	@Override
	public void setValuesToViews() {
		binding.conditionsGrid.removeAllViews();

		if (currentConditionsDto.getWeatherDescription() != null) {
			addGridItem(R.string.weather, currentConditionsDto.getWeatherDescription(),
					currentConditionsDto.getWeatherIcon());
		}

		if (currentConditionsDto.getPrecipitationType() != null) {
			addGridItem(R.string.precipitation_type, currentConditionsDto.getPrecipitationType(), null);
		}
		if (currentConditionsDto.isHasPrecipitationVolume()) {
			addGridItem(R.string.precipitation_volume, currentConditionsDto.getPrecipitationVolume(), null);
		}
		if (currentConditionsDto.isHasRainVolume()) {
			addGridItem(R.string.rain_volume, currentConditionsDto.getRainVolume(), null);
		}
		if (currentConditionsDto.isHasSnowVolume()) {
			addGridItem(R.string.snow_volume, currentConditionsDto.getSnowVolume(), null);
		}

		if (currentConditionsDto.getTemp() != null) {
			addGridItem(R.string.temperature, currentConditionsDto.getTemp(),
					null);
		}
		if (currentConditionsDto.getFeelsLikeTemp() != null) {
			addGridItem(R.string.real_feel_temperature, currentConditionsDto.getFeelsLikeTemp(),
					null);
		}
		if (currentConditionsDto.getHumidity() != null) {
			addGridItem(R.string.humidity, currentConditionsDto.getHumidity(), null);
		}
		if (currentConditionsDto.getWindDirection() != null) {
			View windDirectionView = addGridItem(R.string.wind_direction, currentConditionsDto.getWindDirection(),
					R.drawable.arrow);
			((ImageView) windDirectionView.findViewById(R.id.label_icon)).setRotation(currentConditionsDto.getWindDirectionDegree() + 180);
		}
		if (currentConditionsDto.getWindSpeed() != null) {
			addGridItem(R.string.wind_speed,
					currentConditionsDto.getWindSpeed(), null);
		}
		if (currentConditionsDto.getWindGust() != null) {
			addGridItem(R.string.wind_gust,
					currentConditionsDto.getWindGust() == null ? getString(R.string.not_available) :
							currentConditionsDto.getWindGust(), null);
		}
		if (currentConditionsDto.getWindStrength() != null) {
			addGridItem(R.string.wind_strength, currentConditionsDto.getSimpleWindStrength(), null);
		}
		if (currentConditionsDto.getPressure() != null) {
			addGridItem(R.string.pressure, currentConditionsDto.getPressure(), null);
		}
		if (currentConditionsDto.getVisibility() != null) {
			addGridItem(R.string.visibility, currentConditionsDto.getVisibility(), null);
		}
		if (currentConditionsDto.getCloudiness() != null) {
			addGridItem(R.string.cloud_cover, currentConditionsDto.getCloudiness(), null);
		}
		if (currentConditionsDto.getDewPoint() != null) {
			addGridItem(R.string.dew_point, currentConditionsDto.getDewPoint(), null);
		}
		if (currentConditionsDto.getUvIndex() != null) {
			addGridItem(R.string.uv_index, currentConditionsDto.getUvIndex(), null);
		}
	}

	public void setCurrentConditionsDto(CurrentConditionsDto currentConditionsDto) {
		this.currentConditionsDto = currentConditionsDto;
	}
}

File: app/src/main/java/com/lifedawn/bestweather/ui/weathers/detail/currentconditions/DetailCurrentConditionsFragment.kt
package com.lifedawn.bestweather.ui.weathers.detail.currentconditions

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import com.lifedawn.bestweather.R
import com.lifedawn.bestweather.commons.constants.WeatherDataType
import com.lifedawn.bestweather.data.local.weather.models.CurrentConditionsDto
import com.lifedawn.bestweather.data.local.weather.models.DailyForecastDto.Values.isHasPrecipitationVolume
import com.lifedawn.bestweather.data.local.weather.models.DailyForecastDto.Values.isHasRainVolume
import com.lifedawn.bestweather.data.local.weather.models.DailyForecastDto.Values.isHasSnowVolume
import com.lifedawn.bestweather.ui.weathers.detail.base.BaseDetailCurrentConditionsFragment

class DetailCurrentConditionsFragment : BaseDetailCurrentConditionsFragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        currentConditionsDto = bundle!!.getSerializable(WeatherDataType.currentConditions.name) as CurrentConditionsDto?
        setValuesToViews()
    }

    override fun onResume() {
        super.onResume()
        weatherFragmentViewModel!!.onResumeWithAsync(this)
    }

    override fun setValuesToViews() {
        binding!!.conditionsGrid.removeAllViews()
        if (currentConditionsDto!!.weatherDescription != null) {
            addGridItem(
                R.string.weather, currentConditionsDto!!.weatherDescription,
                currentConditionsDto.getWeatherIcon(), true
            )
        }
        if (currentConditionsDto!!.precipitationType != null) {
            addGridItem(R.string.precipitation_type, currentConditionsDto!!.precipitationType, null)
        }
        if (currentConditionsDto.isHasPrecipitationVolume()) {
            addGridItem(R.string.precipitation_volume_of_grid, currentConditionsDto!!.precipitationVolume, null)
        }
        if (currentConditionsDto.isHasRainVolume()) {
            addGridItem(R.string.rain_volume_of_grid, currentConditionsDto!!.rainVolume, null)
        }
        if (currentConditionsDto.isHasSnowVolume()) {
            addGridItem(R.string.snow_volume_of_grid, currentConditionsDto!!.snowVolume, null)
        }
        if (currentConditionsDto!!.temp != null) {
            addGridItem(
                R.string.temperature, currentConditionsDto!!.temp,
                null
            )
        }
        if (currentConditionsDto!!.feelsLikeTemp != null) {
            addGridItem(
                R.string.wind_chill_temperature_of_grid, currentConditionsDto!!.feelsLikeTemp,
                null
            )
        }
        if (currentConditionsDto!!.humidity != null) {
            addGridItem(R.string.humidity, currentConditionsDto!!.humidity, null)
        }
        if (currentConditionsDto!!.windDirection != null) {
            val windDirectionView = addGridItem(
                R.string.wind_direction, currentConditionsDto!!.windDirection,
                R.drawable.arrow, false
            )
            (windDirectionView.findViewById<View>(R.id.label_icon) as ImageView).rotation =
                (currentConditionsDto!!.windDirectionDegree + 180).toFloat()
        }
        if (currentConditionsDto!!.windSpeed != null) {
            addGridItem(
                R.string.wind_speed,
                currentConditionsDto!!.windSpeed, null
            )
        }
        if (currentConditionsDto!!.windGust != null) {
            addGridItem(
                R.string.wind_gust,
                if (currentConditionsDto!!.windGust == null) getString(R.string.not_available) else currentConditionsDto!!.windGust, null
            )
        }
        if (currentConditionsDto!!.windStrength != null) {
            addGridItem(R.string.wind_strength, currentConditionsDto!!.simpleWindStrength, null)
        }
        if (currentConditionsDto!!.pressure != null) {
            addGridItem(R.string.pressure, currentConditionsDto!!.pressure, null)
        }
        if (currentConditionsDto!!.visibility != null) {
            addGridItem(R.string.visibility, currentConditionsDto!!.visibility, null)
        }
        if (currentConditionsDto!!.cloudiness != null) {
            addGridItem(R.string.cloud_cover, currentConditionsDto!!.cloudiness, null)
        }
        if (currentConditionsDto!!.dewPoint != null) {
            addGridItem(R.string.dew_point, currentConditionsDto!!.dewPoint, null)
        }
        if (currentConditionsDto!!.uvIndex != null) {
            addGridItem(R.string.uv_index, currentConditionsDto!!.uvIndex, null)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        currentConditionsDto = null
    }

    fun setCurrentConditionsDto(currentConditionsDto: CurrentConditionsDto) {
        this.currentConditionsDto = currentConditionsDto
    }

    companion object {
        private var currentConditionsDto: CurrentConditionsDto? = null
    }
}